syntax = "proto3";

message Ps2State {
  uint32    status          = 1;
  uint32    mode            = 2;

  message Mouse {
    uint32  buttons         = 1;
    uint32  dx              = 2;
    uint32  dy              = 3;
    uint32  resolution      = 4;
    uint32  sample_rate     = 5;
    uint32  scaling         = 6;
    uint32  stream_mode     = 7;
  }

  message Keyboard {
    uint32  scancode_set    = 1;
    uint32  leds            = 2;
  }
  
  Keyboard  keyboard        = 3;
  Mouse     mouse           = 4;
}
